tests/net/wifi/configs/wifi.build.* fail due to build warnings in native_sim//64

**Describe the bug**
tests/net/wifi/configs/wifi.build.* fail in twister due to a lot of trivial build warnings in native_sim/native/64


**To Reproduce**
Steps to reproduce the behavior:
1. twister -p native_sim/native/64 -T tests/net/wifi/
4. See errors

**Expected behavior**
No build warnings & therefore no twister failures

**Impact**
Theses tests failing in 64 bit platforms

**Logs and console output**
A few tenths of warnings like
```
modules/lib/hostap/hostapd/hostapd_cli_zephyr.c:269:47: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘size_t’ {aka ‘long unsigned int’} [-Wformat=]
  269 |                         wpa_printf(MSG_DEBUG, "Received len: %d, msg_len:%d - %s->END\n",
      |                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
A couple like
```
modules/lib/hostap/wpa_supplicant/wpa_cli_zephyr.c:506:30: error: passing argument 1 of ‘supp_make_argv’ from incompatible pointer type [-Wincompatible-pointer-types]
  506 |         (void)supp_make_argv(&argc, &argv[0], cmd, MAX_ARGS);
      |                              ^~~~~
      |                              |
      |                              int *
```
And 
```
modules/lib/hostap/src/ap/ap_drv_ops.c:272:41: warning: overflow in conversion from ‘long unsigned int’ to ‘int’ changes value from ‘18446744073709551614’ to ‘-2’ [-Woverflow]
  272 |                                      0, ~WPA_STA_AUTHORIZED);
```

**Environment (please complete the following information):**

 - OS: Ubuntu Linux 24.04
 - Toolchain (e.g Zephyr SDK, ...): Host gcc 14.2
 - Commit SHA or Version used: Main as of now 95cc5f53b82cbc87da1e0f8f402bb1430bda9d66 

**Additional context**

--- a/tests/net/wifi/configs/testcase.yaml
+++ b/tests/net/wifi/configs/testcase.yaml
@@ -6,7 +6,6 @@ common:
     - net
   platform_allow:
     - native_sim
-    - native_sim/native/64
 tests:
   wifi.build.crypto_default:
     extra_configs:

